go.mod

```diff
@@ -1,24 +1,26 @@
 module github.com/rancher/rke
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.3
```

Comment:
`Toolchain` is new to me, go1.22.3 seems to be for Go 1.22.3, not 1.22.0, so could you explain why it was chosen?

Reply:
Afaik, toolchain version is required >= go version so v1.22.3 should be ok but good question in case there was something specific that needed 1.22.3 instead of latest 1.22.4

Reply:
@kinarashah Thank you for the explanation. 

> A module’s go line must declare a version greater than or equal to the go version declared by each of the modules listed in require statements. 

Based on the above statement from the [Golang doc](https://go.dev/doc/toolchain), and the fact that rancher/rancher declares go1.22.3 ([code](https://github.com/rancher/rancher/blob/release/v2.9/go.mod#L5C11-L5C19)), it makes seems to use go1.22.3 here.